src/olympia/devhub/tasks.py

```diff
@@ -351,8 +351,9 @@ def annotate_legacy_addon_restrictions(results, is_new_upload):
         max_target_firefox_version < 99000000000000)
 
     # Thunderbird/Seamonkey only add-ons are moving to addons.thunderbird.net.
-    if (waffle.switch_is_active('disallow-thunderbird-and-seamonkey') and
-            is_targeting_thunderbird_or_seamonkey_only):
+    if (not is_webextension and
```

Comment:
Lets hope no-one tries to upload a thunderbird only webextension https://bugzilla.mozilla.org/show_bug.cgi?id=1396172 😉

Reply:
what's this needed for anyway - `is_targeting_thunderbird_or_seamonkey_only` should be `False` regardless of the tech used for the addon... right? Or don't webextensions have correct apps?

Reply:
Re-reading this... yeah, I'm not sure how we ever ended up in that case. I'll revisit this.

Reply:
It's because when the add-on is parsed with the linter, `metadata['applications']` is absent - the linter cannot determine this, since webextensions manifests don't contain actual applications information, only "engines".

So `target_apps` was empty, and that was causing the issue. Adding the check on `target_apps` in both variables like I did is enough so we don't need the extra `is_webextension` check. I could add an early return at the beginning of the function if `is_webextension` is `True`, though: after all, this is supposed to be `annotate_legacy_addon_restrictions` ...

Reply:
4fac771ae1d858746cf110f7b989e8fe7538edbd